src/features/click-to-play.js

```diff
@@ -1411,17 +1403,32 @@ export function init (args) {
     })
 
     // Request the current state of Click to Load from the platform.
-    // Note: When the response is received, initCTL() is then called by the
-    //       response handler to finish starting up the feature.
+    // Note: When the response is received, the response handler finishes
+    //       starting up the feature.
     sendMessage('getClickToLoadState')
 }
 
-export function update (args) {
-    const detail = args && args.detail
-    if (!(detail && detail.func)) { return }
+export function update (message) {
+    // TODO: Once all Click to Load messages include the feature property, drop
+    //       messages that don't include the feature property too.
+    if (message?.feature && message?.feature !== 'clickToLoad') return
```

Comment:
Update "messages" are sent to all enabled features, so messages of one feature could clash with another. I made a start adding a feature property here, but in the future we should ensure it's sent more consistently.

Reply:
Yea, if the content-scope-scripts abstractions takes care of this in the future then it sounds good to me. I just thought I'd keep the changes for this PR minimal, while making the Click to Load messages make sense (since ports are in progress).

Reply:
It's only on the extension platforms currently and you're changing out how all the messaging is working.

Maybe I'm misunderstanding the complexity here but we can probably move this if statement up a level to only trigger to the correct feature in this PR safely. (I've not looked at update in a while so might be totally misremembering)

Anyway I think it's serious enough to scope as a task before we end up with a runtime issues overtriggering etc.

Reply:
We can't move the feature check up a level yet unfortunately, since message responses don't include the "feature" property. I started looking at including feature in message resposnes, but then the original message needed to include the "feature" property too, and to add those we needed to change all Click to Load messages to use Objects for the message payload so the "feature" could be included. Like I say, with this PR I'm trying to keep the changes minimal while making the API make sense for folks porting Click to Load to other platforms.